tesseract_monitoring/src/robot_comm_monitor_node.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * robot_communication_monitor.cpp
+ *
+ * Checks for fanuc communication errors
+ *
+ * Created on June 14, 2019
+ *    Author: rwall
+ *
+ *
+ */
+
+#include "boost/bind.hpp"
+#include "ros/ros.h"
+#include "std_msgs/Bool.h"
+#include "std_msgs/String.h"
+#include <sensor_msgs/JointState.h>
+#include <tesseract_monitoring/current_state_monitor.h>
+
+// Test run commands:
+// rosrun tesseract_monitoring robot_communication_monitor_node
+// rostopic pub -r 1 /joint_states sensor_msgs/JointState
+// rostopic echo /fanuc_com_error
+
+class FanucCommError
+{
+public:
+  FanucCommError(ros::NodeHandle& nh)
+  {
+    fanuc_comm_error.data = false;
+    chatter_pub = nh.advertise<std_msgs::Bool>("fanuc_com_error", 20);
+
+    chatter_sub = nh.subscribe("joint_states", 20, &FanucCommError::subscriberCallback, this);
+  }
+  void startTimer()
+  {
+    timer = FanucCommError::nh.createTimer(ros::Duration(error_timer), &FanucCommError::timerCallback, this);
+  }
+
+  void subscriberCallback(const sensor_msgs::JointStateConstPtr& joint_state)
+  {
+    count = count + 1;  // increase count of joint states subscription
+
+    // ROS_INFO("\n subscriber called. Count is %d \n", count); //debugging
+    // statement
+  }
+
+  void timerCallback(const ros::TimerEvent&)
+  {
+    // ROS_INFO("\n Timer called. Count is %d",count); //debugging statement
+
+    if (count < 1)
+    {
+      fanuc_comm_error.data = true;
+    }
+
+    else
+    {
+      fanuc_comm_error.data = false;
+    }
+
+    count = 0;                              // re-initialize error check
+    chatter_pub.publish(fanuc_comm_error);  // publish error status
+
+    // ROS_INFO("FANUC COM ERROR= %s \n", fanuc_comm_error.data ? "true" :
+    // "false");//debugging statement
+  }
+
+private:
+  ros::Timer timer;
```

Comment:
I know it's a pain, but ROS style is to have trailing underscores on class member variables like `ros::Timer timer_`. See [HERE](http://wiki.ros.org/CppStyleGuide#Classes_.2BAC8_Types)